src/results.cpp

```diff
@@ -488,6 +503,9 @@ Query Results::get_query() const
             if (query.get_table()) {
                 return query;
             }
+            if (m_table_view.is_backlink_view()) {
+                return Query();
```

Comment:
This does need to return a real query as we support filtering and sorting backlink views.

Reply:
I realize that this was just a cover for underlying problems. Made some fixes in core (setting `m_view` when importing) and results_notifier.cpp (setting of `m_results_were_used`)